src/python/pants/option/arg_splitter.py

```diff
@@ -23,7 +22,9 @@ class ArgSplitterError(Exception):
 class SplitArgs:
     """The result of splitting args."""
 
+    builtin_goals: list[str]  # Requested builtin goals (explicitly or implicitly).
     goals: list[str]  # Explicitly requested goals.
+    unknown_goals: list[str]  # Any unknown goals.
```

Comment:
We need to expose these now, in order for built in goals to be able to act upon them.